Malmo/src/ClientConnection.cpp

```diff
@@ -28,85 +28,117 @@
+    boost::shared_ptr<ClientConnection> ClientConnection::create(boost::asio::io_service& io_service, std::string address, int port)
     {
-        return boost::shared_ptr< ClientConnection >(new ClientConnection( io_service, address, port ) );
+        return boost::shared_ptr<ClientConnection>(new ClientConnection(io_service, address, port));
     }
-    
-    void ClientConnection::send( std::string message )
+
+    void ClientConnection::send(std::string message)
     {
-        //LOGTRACE(LT("Request to send "), message, LT(" to "), this->)
-        this->io_service.post( boost::bind( &ClientConnection::writeImpl, shared_from_this(), message ) );
+        this->io_service.post(boost::bind(&ClientConnection::writeImpl, shared_from_this(), message));
     }
 
-    ClientConnection::ClientConnection( boost::asio::io_service& io_service, std::string address, int port )
-        : io_service( io_service )
+    ClientConnection::ClientConnection(boost::asio::io_service& io_service, std::string address, int port)
+        : io_service(io_service)
     {
+        // DEBUG FAKE 
+        // std::cout << "CONNECT SIT DOWN" << std::endl;
+        // std::this_thread::sleep_for(std::chrono::seconds(1000));
+
+
         LOGTRACE(LT("Creating ClientConnection to "), address, LT(":"), port);
-        // connect the socket to the requested endpoint
-        boost::asio::ip::tcp::resolver resolver( io_service );
-        boost::asio::ip::tcp::resolver::query query( address, std::to_string( port ) );
-        boost::system::error_code ec;
-        boost::asio::ip::tcp::resolver::iterator endpoint_iterator = resolver.resolve( query, ec );
-        if (ec)
-            LOGERROR(LT("Failed to resolve endpoint "), address, LT(":"), port, LT(" - "), ec.message());
-        try
-        {
-            this->socket = std::unique_ptr< boost::asio::ip::tcp::socket >(new boost::asio::ip::tcp::socket(io_service));
-            boost::asio::ip::tcp::resolver::iterator connected_endpoint = boost::asio::connect(*this->socket, endpoint_iterator);
-            LOGFINE(LT("Connected - "), connected_endpoint->service_name(), LT(" - "), connected_endpoint->host_name());
-        }
-        catch (boost::system::system_error e)
-        {
-            LOGERROR(LT("Failed to connect to "), address, LT(":"), port, LT(" - "), e.code().message());
-        }
+
+        resolver = std::unique_ptr<boost::asio::ip::tcp::resolver>(new boost::asio::ip::tcp::resolver(io_service));
+        query = std::unique_ptr<boost::asio::ip::tcp::resolver::query>(new boost::asio::ip::tcp::resolver::query(address, std::to_string(port)));
+
+        socket = std::unique_ptr<boost::asio::ip::tcp::socket>(new boost::asio::ip::tcp::socket(io_service));
+        deadline = std::unique_ptr<boost::asio::deadline_timer>(new boost::asio::deadline_timer(io_service, boost::posix_time::seconds(60)));
```

Comment:
Tiny nitpick - I saw the Rpc class has the timeout as a member, whereas here it's hard-coded.

Reply:
I'll change to follow the same pattern.